### DISCUSSION TITLE
Getting rid of the left hand menu when displaying OpenAPI spec content

### DISCUSSION THREAD

**Discussion by alecthegeek** [thegeeklab/hugo-geekdoc#122] [Q&A] [answered]
I have a question about how to remove the left hand menu in certain pages

My API docs are in OpenAPI.

I have used `redo-cli bundle` to create HTML (and `sed` to add the front matter)

However the OpeAPI content is very wide (3 columns) and it would be very useful to switch off the left hand nav menu on this one page.

Is there some cleaver shortcode or front matter content I can use to switch this off please?

BTW, the issue is the same if you use Swagger

**Comment by xoxys** [comment]
Hmm should be possible by a page option, but I'm not sure if I like the idea as it breaks the fundamental design a bit.

**Reply by alecthegeek** [reply]
I agree about the design, but I am struggling to think of a better way to display the three column content of the API reference.

**Reply by xoxys** [reply]
Let's do it, I'll look into this tomorrow.

**Reply by alecthegeek** [reply]
let me know what I can do to help (I am Melbourne Australia)

**Reply by xoxys** [reply]
Added if https://github.com/thegeeklab/hugo-geekdoc/issues/128

**Comment by alecthegeek** [comment]
Here is a screen shot that might help show the problem

![image](https://user-images.githubusercontent.com/14003/118575318-8a357200-b7c9-11eb-9f82-3e867518432b.png)

**Comment by xoxys** [comment, marked as answer]
I've added this feature in the latest release. Would be awesome if you could give it a test. I've also restructured the landing page of https://geekdocs.de/ to demo it.

**Reply by alecthegeek** [reply]
🥳 

Can't wait to try it. Will do it today.

**Reply by alecthegeek** [reply]
Big improvement

![image](https://user-images.githubusercontent.com/14003/118735508-f1aff800-b883-11eb-9768-a3a507486fc8.png)

Thank you so much.

**Comment by alecthegeek** [comment]
To help people to use this, this is my basic workflow using `redoc-cli`

1. Install redoc-cli via npm
2. Before running Hugo do the following

```
cd content/API-reference/ &&
/build/node_modules/.bin/redoc-cli bundle pmitc-api.yaml && 
sed -i -e '1i ---\
title: "API Reference"\
draft: false\
weight: 90\
geekdocNav: false\
geekdocAlign: left\
---' redoc-static.html &&
mv redoc-static.html _index.html 
cd $OLDPWD
```

Obvs you will need to change the file and directory names to suite

The entry in the menu YAML file is

```
  - name: "OpenAPI Reference"
    ref: "/API-reference/"
```
